src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Reindex/ReindexUtilities.cs

```diff
@@ -91,30 +91,14 @@ public async Task ProcessSearchResultsAsync(SearchResult results, IReadOnlyDicti
 
         public async Task<(bool, string)> UpdateSearchParameters(IReadOnlyCollection<string> searchParameterUris, CancellationToken cancellationToken)
         {
-            var searchParameterStatusList = new List<ResourceSearchParameterStatus>();
-
-            foreach (string uri in searchParameterUris)
+            try
             {
-                var searchParamUri = new Uri(uri);
-
-                try
-                {
-                    _searchParameterDefinitionManager.SetSearchParameterEnabled(searchParamUri);
-                }
-                catch (SearchParameterNotSupportedException)
-                {
-                    return (false, string.Format(Core.Resources.SearchParameterNoLongerSupported, uri));
-                }
-
-                searchParameterStatusList.Add(new ResourceSearchParameterStatus()
-                {
-                    LastUpdated = DateTimeOffset.UtcNow,
-                    Status = SearchParameterStatus.Enabled,
-                    Uri = searchParamUri,
-                });
+                await _searchParameterStatusManager.UpdateSearchParameterStatus(searchParameterUris, SearchParameterStatus.Enabled);
+            }
+            catch (SearchParameterNotSupportedException)
+            {
+                return (false, string.Format(Core.Resources.SearchParameterNoLongerSupported));
```

Comment:
>SearchParameterNoLongerSupported [](start = 60, length = 32)

Can we take name from SearchParameterNotSupportedException and pass it to error message? It can be confusing from user perspective to do binary search on list of params to find out which one of them is not working.

Reply:
Updated.